### PR TITLE
Refactoring/adding tests

### PR DESCRIPTION
**Pull request by zfhui**
* added RSpec 
* added tests
* update .gitignore

ping @edenspiekermann/backend

### CODE REVIEW

**Review by eschaefer — APPROVED** [review]
👍

File: lib/kupong_integration/service.rb [hunk 1]
```diff
@@ -95,11 +95,11 @@ def sanitize(phone)
     end
 
     def authorization
-      settings[:authorization]
+      settings[:authorization] || settings['authorization']
```
Comment (nburkley) [inline lib/kupong_integration/service.rb:98]: Would it bet better convert the settings hash using `hash_with_indifferent_access` on initialize so we don't have to do this here and for `coupon_id`?

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  #     --seed 1234
+  config.order = :random
+
+  # Seed global randomization in this process using the `--seed` CLI option.
+  # Setting this allows you to use `--seed` to deterministically reproduce
+  # test failures related to randomization by passing the same `--seed` value
+  # as the one that triggered the failure.
+  Kernel.srand config.seed
+=end
+end
```
Comment (nburkley) [inline spec/spec_helper.rb:103]: You could probably just remove all the getting started boiler plate comments

File: lib/kupong_integration/service.rb (hunk 1 above)
Comment (zfhui) [reply lib/kupong_integration/service.rb:98]: That's part of ActiveSupport, so it's not available here.

File: spec/spec_helper.rb (hunk 2 above)
Comment (zfhui) [reply spec/spec_helper.rb:103]: Yes, will do this in the next PR.

**Pull request merged**